irods/session.py

```diff
@@ -162,14 +167,64 @@ def __del__(self):
         if self.pool is not None:
             self.cleanup()
 
-    def cleanup(self):
-        for conn in self.pool.active | self.pool.idle:
-            try:
-                conn.disconnect()
-            except NetworkException:
-                pass
-            conn.release(True)
+    @staticmethod
+    def type_to_str(t):
+        try:
+            return t.__name__
+        except:
+            return str(t)
+
+    def _list(self,obj,seen=(),indent=0):
+        seen = (seen if isinstance(seen,dict) else dict(seen))
+        seen[obj] = True
+        for k,v in vars(obj).items():
+            indent_ = ' '*indent
+            if isinstance(v,iRODSSession) and v is not self: 
+                indent = '[E]'+indent[3:]
+            print(indent_, end='')
+            #print(k, self.type_to_str(type(v)), v, sep='\t')
+            print('%-20s %-20s %-50s'%(k, self.type_to_str(type(v)), v))
```

Comment:
debugging?

Reply:
The `list` function was for debugging primarily. Still wonder if it may have permanent utility.  I suspect it will disappear here, but persist in a different dev branch for the benefit of debugging irods.experimental.client.simple.Session for some time yet....